src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt

```diff
@@ -259,4 +259,20 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
       " ORDER BY v.sessionSlot.slotStart,v.id",
   )
   fun findBookedVisitsForDate(prisonCode: String, date: LocalDate): List<Visit>
+
+  @Query(
+    "Select v.reference FROM visit v " +
+      " LEFT JOIN event_audit ea on ea.booking_reference = v.reference and ea.type in ('UPDATED_VISIT','BOOKED_VISIT') " +
+      " LEFT JOIN actioned_by ab on ab.id = ea.actioned_by_id" +
+      " LEFT JOIN session_slot ss on ss.id = v.session_slot_id " +
+      " WHERE ab.booker_reference = :bookerReference AND v.visit_status = 'BOOKED' AND ss.slot_date >= CURRENT_DATE AND " +
```

Comment:
do we need to compare on time as well? ss.slot_start?

Reply:
may be....

Reply:
Leave it for now, but could change to solt_end to incl current visits